tools/congestion-model/src/evaluation/mod.rs

```diff
@@ -72,4 +76,53 @@ impl Model {
             failed_transactions,
         }
     }
+
+    pub fn write_stats_header(&self, stats_writer: &mut StatsWriter) {
+        let Some(stats_writer) = stats_writer else { return };
+
+        stats_writer.write_field("time").unwrap();
+        stats_writer.write_field("round").unwrap();
+
+        stats_writer.write_field("finished_transactions").unwrap();
+        stats_writer.write_field("pending_transactions").unwrap();
+        stats_writer.write_field("waiting_transactions").unwrap();
+        stats_writer.write_field("failed_transactions").unwrap();
+
+        for shard_id in self.shard_ids.clone() {
+            for queue in self.queues.shard_queues(shard_id) {
+                let field_name = format!("shard_{}_queue_{}", shard_id, queue.name());
+                stats_writer.write_field(field_name).unwrap();
+            }
+        }
+        stats_writer.write_record(None::<&[u8]>).unwrap();
+    }
+
+    pub fn write_stats_values(
+        &self,
+        stats_writer: &mut StatsWriter,
+        start_time: chrono::prelude::DateTime<Utc>,
+        round: usize,
+    ) {
+        let Some(stats_writer) = stats_writer else { return };
+
+        let time = start_time + Duration::seconds(round as i64);
+
+        stats_writer.write_field(format!("{:?}", time)).unwrap();
+        stats_writer.write_field(format!("{}", round)).unwrap();
+
+        // This is slow and takes up to 10s for the slowest workloads and strategies.
+        let progress = self.progress();
```

Comment:
@jakmeier Let me know if this is something to be addressed. I think it's fine because it's still ok to run a single strategy and workload and you'd typically want the stats only at the end of dev cycle. One way to improve this would be to add sampling.